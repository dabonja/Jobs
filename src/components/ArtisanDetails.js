import { getDefaultNormalizer } from '@testing-library/dom';
import React from 'react';
import mechanic from '../icons/mechanic.png';
const ArtisanDetails = ({ selectedArtisan }) => {

    return (
        <div className="d-flex justify-content-center" style={{ position: 'relative', top: '8rem' }}>
            <div className="card mb-3" style={{ maxWidth: " 70rem" }}>
                <div className="row g-0">
                    <div className="col-md-4">
                        <img src={mechanic} className="img-fluid rounded-start" alt="..." style={{ width: '230rem', height: '20rem' }} />
                    </div>
                    <div className="col-md-8">
                        <div className="card-body">
                            <h5 className="card-title">{selectedArtisan.fullName}</h5>
                            <p className="card-text">Get in touch with this artisan! Number is below this text.</p>
                            <p className="card-text">Phone number: {selectedArtisan.contact}</p>
                            <p className="card-text">Profession: {selectedArtisan.profession}</p>
                            <p className="card-text">Location: {selectedArtisan.location}</p>
                            <p className="card-text"><small className="text-muted">Last updated 3 mins ago</small></p>
                            <p className="card-text"><a href="#" >View all comments and rating</a></p>
                        </div>
                    </div>
                </div>
            </div>
        </div>
    )
}

export default ArtisanDetails;