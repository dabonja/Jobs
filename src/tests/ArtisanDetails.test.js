import ArtisanDetails from '../components/ArtisanDetails';
import renderer from 'react-test-renderer';

it('renders ArtisanDetails component correctly', () => {

    let artisan = {
        company: "Eskobarova radionica",
        contact: 634424242,
        fullName: "Danijel Dabic",
        id: 1,
        image: "image_adress",
        location: "Beska",
        profession: "Web Developer",
        rated: 7,
    }

    const tree = renderer.create(<ArtisanDetails selectedArtisan={artisan}/>)
        .toJSON();
    expect(tree).toMatchSnapshot()
})