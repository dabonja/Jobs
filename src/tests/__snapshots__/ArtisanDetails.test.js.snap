// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ArtisanDetails component correctly 1`] = `
<div
  className="d-flex justify-content-center"
  style={
    Object {
      "position": "relative",
      "top": "10rem",
    }
  }
>
  <div
    className="card mb-3"
    style={
      Object {
        "maxWidth": " 70rem",
      }
    }
  >
    <div
      className="row g-0"
    >
      <div
        className="col-md-4"
      >
        <img
          alt="..."
          className="img-fluid rounded-start"
          src="mechanic.png"
          style={
            Object {
              "height": "20rem",
              "width": "230rem",
            }
          }
        />
      </div>
      <div
        className="col-md-8"
      >
        <div
          className="card-body"
        >
          <h5
            className="card-title"
          >
            Danijel Dabic
          </h5>
          <p
            className="card-text"
          >
            Get in touch with this artisan! Number is below this text.
          </p>
          <p
            className="card-text"
          >
            Phone number: 
            634424242
          </p>
          <p
            className="card-text"
          >
            Profession: 
            Web Developer
          </p>
          <p
            className="card-text"
          >
            Location: 
            Beska
          </p>
          <p
            className="card-text"
          >
            <small
              className="text-muted"
            >
              Last updated 3 mins ago
            </small>
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
